pkg/restapi/operation/didexchange/didexchange.go

```diff
@@ -67,26 +77,105 @@ type Operation struct {
 // Responses:
 //    default: genericError
 //        200: createInvitationResponse
-func (c *Operation) CreateInvitation(rw http.ResponseWriter, req *http.Request, param denco.Params) {
+func (c *Operation) CreateInvitation(rw http.ResponseWriter, req *http.Request) {
 
 	logger.Debugf("Creating connection invitation ")
-
 	// call didexchange client
-	response, err := c.didExchangeClient.CreateInvitation()
+	response, err := c.client.CreateInvitation()
 	if err != nil {
 		c.writeGenericError(rw, err)
 		return
 	}
 
-	err = json.NewEncoder(rw).Encode(&response.Invitation)
+	err = json.NewEncoder(rw).Encode(&models.CreateInvitationResponse{Payload: response})
+	if err != nil {
+		logger.Errorf("Unable to write response, %s", err)
+	}
+}
+
+// ReceiveInvitation swagger:route POST /receive-invitation did-exchange receiveInvitation
+//
+// Receive a new connection invitation....
+//
+// Responses:
+//    default: genericError
+//        200: receiveInvitationResponse
+func (c *Operation) ReceiveInvitation(rw http.ResponseWriter, req *http.Request) {
+
+	logger.Debugf("Receiving connection invitation ")
+
+	payload, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		c.writeGenericError(rw, err)
+		return
+	}
+
+	err = c.service.Handle(dispatcher.DIDCommMsg{Type: didexchange2.ControllerReceiveInvitation, Payload: payload})
```

Comment:
dispatcher.DIDCommMsg.Type should be set to the `Type` of the enclosed `didexchange.Invitation` inside http request `req`

Reply:
done